Graphviz/DOT.sublime-syntax

```diff
@@ -1,62 +1,391 @@
 %YAML 1.2
 ---
-# http://www.sublimetext.com/docs/3/syntax.html
+# http://www.sublimetext.com/docs/syntax.html
+# https://graphviz.org/doc/info/lang.html
+
 name: Graphviz (DOT)
 scope: source.dot
+version: 2
 
 file_extensions:
   - gv
   - dot
+  - DOT
 
 first_line_match: |-
   (?xi:
-    ^ \s* // .*? -\*- .*? \b(dot|graphviz)\b .*? -\*-  # editorconfig
+    {{leading_wspace}} // .*? -\*- .*? \b(dot|graphviz)\b .*? -\*-  # editorconfig
   )
 
+###[ CONTEXTS ]################################################################
+
 contexts:
+  prototype:
+    - include: comments
+
   main:
-    - include: expressions
+    - include: fullline-comments-number-sign
+    - include: declarations
+    - include: attribute-statements
+    - include: node-statements
+    - include: attribute-lists
+    - include: embedded-html
+    - include: literals
+    - include: operators
+
+###[ COMMENTS ]################################################################
+
+  # https://graphviz.org/doc/info/lang.html#comments-and-optional-formatting
+
+  comments:
+    - include: line-comments
+    - include: block-comments
+
+  line-comments:
+    # NOTE: number-sign line comments are not part of prototype
+
+    - match: /{2}
+      scope: punctuation.definition.comment.dot
+      push: line-comment-double-slash-body
+
+  line-comment-double-slash-body:
+    - meta_include_prototype: false
+    - meta_scope: comment.line.double-slash.dot
+
+    - include: eol-pop
+
+  block-comments:
+    - include: empty-block-comments
+    - include: block-comments-slash-star
+
+  empty-block-comments:
+    - match: /\*{2,}/
+      scope: >-
+        comment.block.empty.dot
+        punctuation.definition.comment.dot
+
+  block-comments-slash-star:
+    - match: /\*
+      scope: punctuation.definition.comment.begin.dot
+      push: block-comment-slash-star-body
+
+  block-comment-slash-star-body:
+    - meta_include_prototype: false
+    - meta_scope: comment.block.dot
 
-  expressions:
-    - match: \b((sub|di)?graph)\b\s+((cluster_)?\w+)\b
+    - match: \*/
+      scope: punctuation.definition.comment.end.dot
+      pop: 1
+    - match: '{{leading_wspace}}(\*)(?!/)'
       captures:
-        1: storage.type.dot
+        1: punctuation.definition.comment.dot
+
+  fullline-comments-number-sign:
+    # NOTE: number-sign line comments are not part of prototype
+
+    - match: '{{leading_wspace}}(?=#)'
+      push:
+        - - match: '#'
+            scope: punctuation.definition.comment.dot
+            set: fullline-comment-number-sign-body
```

Comment:
As number sign comments target full lines, it may be acceptable or even desired to also scope leading whitespace `comment` so any applied background color spans whole line and to ensure a "gabless" `comment` scoped region if multiple lines follow each other. It may help plugins like DoxyGen to maintain comments as scoping of trailing `\n` do.


```suggestion
    - match: '{{leading_wspace}}(#)'
      captures:
        1: punctuation.definition.comment.dot
      push: fullline-comment-number-sign-body
```